### PR TITLE
Jenkins with CentOS 7 and 8

### PR DESCRIPTION
**Pull request by ArnauSistach**
Add Centos 7-specific Dockerfile. The latest version is now CentOS 8, but CentOS 7 is still supported.
It is the same dockerfile as centos with the "from" of CentOS 7.

I would like to have continuity in CentOS 7 `--start-after 2,199`
```
$ docker run --rm -ti jenkins/jenkins:2.198-centos cat /etc/centos-release
CentOS Linux release 7.7.1908 (Core)

$ docker run --rm -ti jenkins/jenkins:2.199-centos cat /etc/centos-release
CentOS Linux release 8.0.1905 (Core)
```

### CODE REVIEW

File: Dockerfile-centos7 [hunk 1]
```diff
@@ -0,0 +1,80 @@
+FROM centos:7
+
+RUN yum update -y && yum install -y epel-release && yum install -y git curl dpkg java java-devel unzip && yum clean all
+
```
Comment (slide) [inline Dockerfile-centos7]: Can you delete any package files that were downloaded to reduce the size of the image?

Comment (ArnauSistach) [reply Dockerfile-centos7]: Need `epel-release` for can install `dpkg`. `dpkg` is for output [`dpkg --print-architecture`](https://github.com/jenkinsci/docker/blob/108fa62a5b299107951a305d5bd3b9e7fed1c4ea/Dockerfile-centos#L39-L40).
Is possible all cases the output is `amd64` and i can modify whit amd64?

- java for funning jenkins.

And this others packages is for passing tests `make test-centos7`
- git
- java-devel
- unzip

Comment (slide) [reply Dockerfile-centos7]: I'm not asking to.remove the packages, just the rpm's or whatever package file format centos uses.

**Comment by ArnauSistach** [comment]
@slide, your merge into centos7 need modify the [labels in Jenkinsfile](https://github.com/jenkinsci/docker/blob/9d2b6d7ed6138a4f9b1203b4dba320dcc38fe242/Jenkinsfile#L34).
Because exists in the commit exists the files [Dockerfile-openj9](https://github.com/jenkinsci/docker/blob/9d2b6d7ed6138a4f9b1203b4dba320dcc38fe242/Dockerfile-openj9) and [Dockerfile-openj9-jdk11](https://github.com/jenkinsci/docker/blob/master/Dockerfile-openj9-jdk11) and modify the Makefile, but the last need is the labels not?

File: Dockerfile-centos7 (hunk 1 above)
Comment (ArnauSistach) [reply Dockerfile-centos7]: If you don't want RPMs, how do you want it done?
The advantage of using RPMs is that it is updated without adding more work.

Comment (slide) [reply Dockerfile-centos7]: I want you to install packages using rpms, but I want the .rpm files removed after the package is installed. If you look at the debian based images, you can see that they remove the .deb files once the packages are installed (https://github.com/jenkinsci/docker/blob/master/Dockerfile#L3). This is common in docker images to make them smaller.

**Comment by slide** [comment]
I merged from master to take care of conflicts because the openj9 PR modified the same lines as this PR.

**Comment by ArnauSistach** [comment]
now why wait to merge?

**Pull request merged**